juniper/src/parser/utils.rs

```diff
@@ -52,15 +52,15 @@ impl Span {
 
 /// Data structure used to wrap items into a [`Span`].
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-pub struct Spanning<T> {
+pub struct Spanning<T, Sp = Span> {
```

Comment:
The added generic argument probably influences how this is documented, but not sure.

The point of the generic is to avoid making a separate type like `BorrowedSpanning`, but I'm up for improvement suggestions.

The ability to borrow a Span can lead to slightly more performant code in various places. Currently only in `look_ahead`, other future suggestions are e.g. the input validation rules.